src/js/tech/ChromecastTech.js

```diff
@@ -0,0 +1,364 @@
+'use strict';
+
+var ChromecastSessionManager = require('../chromecast/ChromecastSessionManager'),
+    ChromecastTechUI = require('./ChromecastTechUI'),
+    _ = require('underscore'),
+    SESSION_TIMEOUT = 10 * 1000, // milliseconds
+    ChromecastTech;
+
+
+ChromecastTech = {
+   constructor: function(options) {
+      var subclass;
+
+      this._eventListeners = [];
+
+      // We have to initialize the UI here, before calling super.constructor
+      // because the constructor calls `createEl`, which references `this._ui`.
+      this._ui = new ChromecastTechUI();
+      this._ui.updatePoster(options.source.poster);
+
+      // Call the super class' constructor function
+      subclass = this.constructor.super_.apply(this, arguments);
+
+      this._chromecastSessionManager = options.source.chromecastSessionManager;
+      this._remotePlayer = this._chromecastSessionManager.getRemotePlayer();
+      this._remotePlayerController = this._chromecastSessionManager.getRemotePlayerController();
+      this._listenToPlayerControllerEvents();
+      this.on('dispose', this._removeAllEventListeners.bind(this));
+
+      this._hasPlayedAnyItem = false;
+      this._currentSource = options.source;
+      this._requestTitle = options.requestTitleFn || _.noop;
+      this._requestSubtitle = options.requestSubtitleFn || _.noop;
+      // See `currentTime` function
+      this._firstItemStartTime = options.startTime || 0;
+      // See `duration` function
+      this._firstItemDuration = options.source.duration || 0;
```

Comment:
Oh, I think I see what you mean now. That won't work because only the first source passed into `options` has a `duration` property because I set it here: https://github.com/silvermine/videojs-chromecast/pull/1/files#diff-0a9db042cf47e5c864efe073284c1ab4R83

Other sources set with `setSource` won't have that property. We have to get it from the `Chromecast` `RemotePlayer` instead.